src/SourceBrowser/src/BinLogToSln/Program.cs

```diff
@@ -290,18 +293,28 @@ static void Main(string[] args)
                 project.WriteLine("  <ItemGroup>");
                 foreach (CommandLineReference reference in invocation.Parsed.MetadataReferences)
                 {
-                    string path = reference.Reference;
-                    if (!File.Exists(path))
+                    includeReference("ReferencePath", reference.Reference);
+                }
+                project.WriteLine("  </ItemGroup>");
+
+                // Add generated files.
+                project.WriteLine("  <ItemGroup>");
+                foreach (var generatedFile in getGeneratedFiles())
+                {
+                    string filePath = generatedFile.FilePath;
+                    if (!File.Exists(filePath))
                     {
-                        Console.WriteLine($"Could not find reference '{path}'");
-                        continue;
+                        Directory.CreateDirectory(Path.GetDirectoryName(filePath));
+                        var stream = generatedFile.Stream;
+                        stream.Position = 0;
+                        using var fileStream = File.OpenWrite(filePath);
+                        stream.CopyTo(fileStream);
                     }
-                    string projToRepoPath = Path.GetRelativePath(invocation.ProjectDirectory, repoRoot);
-                    string projToOutputPath = Path.Join(projToRepoPath, "..");
-                    string refPath = DedupeReference(output, path);
-                    project.WriteLine($"    <ReferencePath Include=\"{Path.Join(projToOutputPath, refPath)}\"/>");
+                    includeFile(filePath, out string projectRelativePath, out string link);
+                    project.WriteLine($"    <Compile Include=\"{projectRelativePath}\"{(link != null ? $" Link=\"{link}\"" : "")}/>");

```

Comment:
Do all callers to `includeFile` need to write the item?  If so, maybe we can move that into the function like it is for `includeReference`?